src/corelibs/U2Formats/src/BAMUtils.cpp

```diff
@@ -209,15 +209,10 @@ static samfile_t* samopen_ugene(int fd, const char* mode, const QString& faiUrl,
 static samfile_t* samOpen(const QString& url, const char* samMode, const QString& faiUrl = "", bam_header_t* header = nullptr) {
     QString fileMode = samMode;
     fileMode.replace("h", "");
-    FILE* file = BAMUtils::openFile(url, fileMode);
-    if (file == nullptr) {
-        return nullptr;
-    }
-    samfile_t* samfile = samopen_ugene(fileno(file), samMode, faiUrl, header);
-    if (samfile == nullptr) {
-        closeFileIfOpen(file);
-        return nullptr;
-    }
+    NP<FILE> file = BAMUtils::openFile(url, fileMode);
```

Comment:
Using `NP<FILE>` instead of `FILE*` make it very visible for code reviewer that this pointer is nullable and must be checked before used.

The unchecked access is also be guarded by SAFE_POINT (runtime issue)

I can't find any non-boilerplate way to make this check compile-time.